src/App.jsx

```diff
@@ -1,5 +1,131 @@
-function App() {
-  return <div className="bg-gray-300 min-h-screen">Calculator</div>;
-}
+import { useState } from "react";
+import CustomButton from "./components/CustomButton";
+
+const App = () => {
+  const [firstNumber, setFirstNumber] = useState("");
+  const [secondNumber, setSecondNumber] = useState("");
+  const [operator, setOperator] = useState("");
+  const handlePickNumber = (number) => {
+    if (!firstNumber) {
+      Number(firstNumber) === 0
+        ? setFirstNumber(number)
+        : setFirstNumber(firstNumber + number);
+    } else {
+      if (operator) {
+        Number(secondNumber) === 0
+          ? setSecondNumber(number)
+          : setSecondNumber(secondNumber + number);
+      } else {
+        Number(firstNumber) === 0
+          ? setFirstNumber(number)
+          : setFirstNumber(firstNumber + number);
+      }
+    }
+  };
```

Comment:
I think it could be better to remove the redundant  Number ()  function call in condition check ,  as variables firstName and second number are already assumed to be numbers , this can help to  make the code more concise and easier to read
```suggestion
const handlePickNumber = (number) => {
  if (!firstNumber) {
    setFirstNumber(Number(firstNumber) === 0 ? number : firstNumber + number);
  } else if (operator) {
    setSecondNumber(Number(secondNumber) === 0 ? number : secondNumber + number);
  } else {
    setFirstNumber(Number(firstNumber) === 0 ? number : firstNumber + number);
  }
};
```